.github/workflows/contracts.yml

```diff
@@ -0,0 +1,32 @@
+---
+name: contracts
+on:
+  push:
+jobs:
+  tests:
+    defaults:
+      run:
+        working-directory: ./packages/contracts
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        node-version: [16.x]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v2
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: Install pnpm and install dependencies 
+        uses: pnpm/action-setup@v2.1.0
+        with:
+          version: 6.0.2
+          run_install: true
+      - name: Run tests
+        run: pnpm run test
+        env:
+          # FIXME hardhat currently pre-configured with many more vars than this
+          # it won't load at all unless they are all defined
```

Comment:
yeah, this has been bugging me! will fix, thanks!

Reply:
@holic I was thinking either:

- include nothing in hardhat by default (not very friendly for newbies)
- setup ENV-detecting conditionals in hardhat (complex but friendly for first-timers)

Reply:
Conditionals are what I've done before and it's a little gross but I like that better than an empty config. I wish hardhat did this better.

But also have in my head that I may move this scaffold to Foundry ;)